app/main.py

```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
-
+    return {
+    "line": error["line_number"],
+    "column": error["column_number"],
+    "message": error["text"],
+    "name": error["code"],
+    "source": "flake8"
+    }
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
-
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed"
+    }
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(file_path, errors)
+        for file_path, errors in format_single_linter_file(file_path, errors)
```

Comment:
Something is definitely wrong because of 2 reasons:
1. you don't use your input parameter `linter_report`
2. `format_single_linter_file` returns 1 dict and you cannot iterate through dict using 2 variables **file_path** and **errors**
3. At the end you call `format_single_linter_file(file_path, errors)`. But where do you take file_path and errors? Your function doesn't have these parameters, you have only **linter_report**

So these issues interdependent. When you fix one, the others are resolved :)